src/lando/headless_api/tests/test_automation_api.py

```diff
@@ -710,6 +714,189 @@ def raise_conflict(*args, **kwargs):
     assert job.status == JobStatus.FAILED
 
 
+def _create_split_branches_for_merge(
+    request, scm, repo_path, main_branch="main", feature_branch="feature"
+):
+    subprocess.run(["git", "switch", main_branch], cwd=repo_path, check=True)
+    main_file = _create_git_commit(request, repo_path)
+    main_commit = (
+        subprocess.run(
+            ["git", "rev-parse", "HEAD"], cwd=repo_path, capture_output=True, check=True
+        )
+        .stdout.decode()
+        .strip()
+    )
+
+    subprocess.run(
+        ["git", "switch", "-c", feature_branch, "HEAD^"], cwd=repo_path, check=True
+    )
+    feature_file = _create_git_commit(request, repo_path)
+    feature_commit = (
+        subprocess.run(
+            ["git", "rev-parse", "HEAD"], cwd=repo_path, capture_output=True, check=True
+        )
+        .stdout.decode()
+        .strip()
+    )
+
+    subprocess.run(["git", "switch", main_branch], cwd=repo_path, check=True)
+
+    return main_commit, main_file, feature_commit, feature_file
+
+
+@pytest.mark.parametrize("strategy", [None, "ours", "theirs"])
+@pytest.mark.django_db
+def test_automation_job_merge_onto_success_git(
+    strategy,
+    repo_mc,
+    treestatusdouble,
+    git_automation_worker,
+    monkeypatch,
+    request,
+):
+    repo = repo_mc(SCM_TYPE_GIT)
+    scm = repo.scm
+    scm.push = mock.MagicMock()
+
+    # Create a repo with diverging history
+    main_commit, main_file, feature_commit, feature_file = (
+        _create_split_branches_for_merge(request, scm, repo.system_path)
+    )
+
+    job = AutomationJob.objects.create(
+        status=JobStatus.SUBMITTED,
+        requester_email="test@example.com",
+        target_repo=repo,
+    )
+    AutomationAction.objects.create(
+        job_id=job,
+        action_type="merge-onto",
+        data={
+            "action": "merge-onto",
+            "commit_message": f"Merge test with strategy {strategy}",
+            "strategy": strategy,
+            "target": feature_commit,
+        },
+        order=0,
+    )
```

Comment:
Could we test the parser that converts the request payload (assuming it's what's in `data`) into an AutomationAction in the DB?

Reply:
Maybe as a fixture that takes the payload, and creates the `AutomationJob` and the `AutomationAction`, so we can reuse it across all the action tests.